tests/langchain/test_langchain_model_export.py

```diff
@@ -1558,6 +1566,117 @@ def extract_history(input):
     }
 
 
+def _extract_endpoint_name_from_lc_model(lc_model, dependency_dict) -> Dict[str, str]:
+    if type(lc_model).__name__ == type(get_fake_chat_model()).__name__:
+        _assign_value_or_append_to_list(
+            dependency_dict, "fake_chat_model_endpoint_name", lc_model.endpoint_name
+        )
+    return dependency_dict
+
+
+@pytest.mark.skipif(
+    Version(langchain.__version__) < Version("0.0.311"), reason="feature not existing"
+)
+@mock.patch(
+    "mlflow.langchain.databricks_dependencies._extract_dependency_dict_from_lc_model",
+    _extract_endpoint_name_from_lc_model,
+)
+def test_databricks_dependency_extraction_from_lcel_chain():
+    from langchain_core.output_parsers import StrOutputParser
+    from langchain_core.prompts import ChatPromptTemplate
+
+    prompt_1 = ChatPromptTemplate.from_template("tell me a short joke about {topic}")
+    prompt_2 = ChatPromptTemplate.from_template(
+        "compare which joke is better {joke1} or {joke2}. Output the better joke."
+    )
+    model_1 = get_fake_chat_model(endpoint_name="fake-endpoint-1")
+    model_2 = get_fake_chat_model(endpoint_name="fake-endpoint-2")
+    model_3 = get_fake_chat_model(endpoint_name="fake-endpoint-3")
+    output_parser = StrOutputParser()
+
+    chain = prompt_1 | {"joke1": model_1, "joke2": model_2} | prompt_2 | model_3 | output_parser
+
+    with mlflow.start_run():
+        model_info = mlflow.langchain.log_model(chain, "basic_chain")
+
+    langchain_flavor = model_info.flavors["langchain"]
+    assert langchain_flavor["databricks_dependency"] == {
+        "fake_chat_model_endpoint_name": ["fake-endpoint-1", "fake-endpoint-2", "fake-endpoint-3"]
+    }
+
+
+def _extract_databricks_dependencies_from_retriever(retriever, dependency_dict) -> Dict[str, str]:
```

Comment:
I see why we're patching these, but can we test the underlying parsing logic contained in the extractors to be sure that we're capturing the required data for logging, just in case the structures change in langchain and the parsing logic needs to update to accomodate?

Reply:
I tried to instantiate a Databricks LLM or DatabricksEmbeddings or ChatDatabricks, but I find it tricky to mock the dependencies. This task is only scoped as 1.5 pts, so it's hard to have a better test coverage. I can file a follow-up 1-point ticket to add tests on underlying logic. Would this work for you?

Reply:
I can imagine that the mocks for those would be pretty complex. I'm wondering if, in a follow up, the extraction and parsing logic could be refactored to be a bit more testable so that we could pinpoint root causes of issues when (not if) the langchain structures change (not a blocker for this PR, but, as you mentioned, probably a good follow up task)

Reply:
Filed https://databricks.atlassian.net/browse/ML-38374